### PR TITLE
Annulus-114

### PR DESCRIPTION
**Pull request by berriose**
I'd like to ask for some input

- suggestion for fixing the error in the second fetching method

### CODE REVIEW

File: lib/blocks/providers/block_provider.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      transactionNumber: 200,
+      withdrawalNumber: 127,
+    );
+    // Adding delay here to simulate API call
+    if (setState) {
+      Future.delayed(
+        Duration(seconds: 1),
+        () {
+          // Do API call here
+          state = const AsyncData(block);
```
Comment (berriose) [inline lib/blocks/providers/block_provider.dart:91]: I'm not sure how to pass in the height since it can't be const. AsyncData is const.

**Comment by berriose** [comment]
I'm going to update the provider to fetch multiple blocks.

File: lib/blocks/providers/block_provider.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import 'package:flutter_annulus/blocks/models/block.dart';
+import 'package:flutter_annulus/chain/models/chains.dart';
+import 'package:flutter_annulus/chain/providers/selected_chain_provider.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+final chainProvider =
```
Comment (KalervoHyyppa) [inline lib/blocks/providers/block_provider.dart:6]: Rename to Block Provider

File: lib/blocks/providers/block_provider.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  /// instance of Genus to target
+  final selectedChain = ref.watch(selectedChainProvider);
+
+  return BlockNotifier(
+    ref,
+    selectedChain,
+  );
+});
+
+class BlockNotifier extends StateNotifier<AsyncValue<Block>> {
```
Comment (KalervoHyyppa) [inline lib/blocks/providers/block_provider.dart:19]: This notifier should return `StateNotifier<AsyncValue<List<Block>>>`, see the latest Transaction Provider for an example

File: lib/blocks/providers/block_provider.dart (hunk 1 above)
Comment (KalervoHyyppa) [reply lib/blocks/providers/block_provider.dart:91]: You can just do `state = AsyncData(transactions);`, and remove the const

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
The biggest change will be updating the Notifiers return value

**Review by KalervoHyyppa — APPROVED** [review]
LGTM

**Pull request merged**